modules/machineset-non-guaranteed-instance.adoc

```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-infrastructure-machinesets.adoc ?
+// * machine_management/creating_machinesets/creating-machineset-aws.adoc
+// * machine_management/creating_machinesets/creating-machineset-azure.adoc (4.6)
+// * machine_management/creating_machinesets/creating-machineset-gcp.adoc (4.6)
+
+[id="machineset-non-guaranteed-instance_{context}"]
+= Creating a spot instance
+
+You can create an AWS MachineSet that deploys machines as non-guaranteed, spot instances to save costs compared to on-demand instance prices. A machine is paired 1:1 with an instance on AWS. The MachineSet ensures the correct number of machines are created. If AWS does not have capacity, the Machine Health Checker (MHC) can remove the machine. The MHC also remediates a machine that fails if a request for a spot instance is not satisfied when a machine is created. AWS does not replace a spot instance when the instance is terminated.
```

Comment:
> A machine is paired 1:1 with an instance on AWS. The MachineSet ensures the correct number of machines are created.

This is generic to MachineSets and not specific to spot instances so probably doesn't need saying here

I think we should have a brief explanation of what a spot instance is somewhere within the docs, ie, you get cheaper instances with the prerequisite agreement that AWS may terminate the instance at any point with a 2 minute warning (would also be good to explain that Openshift will react to this 2 minute warning by starting to drain the workloads from the affected instances)